sdk/azcore/etag.go

```diff
@@ -0,0 +1,80 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcore
+
+import (
+	"strings"
+)
+
+// ETag is a property used for optimistic concurrency during updates
+// ETag is a validator based on https://tools.ietf.org/html/rfc7232#section-2.3.2
+type ETag struct {
+	value *string
```

Comment:
If a just allocate one of these value will be nil; Then, if I call Equals on it, a panic will happen.
So is nil a meaningful etag value or not?
I think we DO need the concept of a missing etag; nil could be this but then Equals needs to be fixed.